graphistry/feature_utils.py

```diff
@@ -93,6 +100,28 @@ def lazy_import_has_min_dependancy():
     except ModuleNotFoundError as e:
         return False, e
 
+def lazy_import_has_dependancy_cuda():
```

Comment:
`_cuda` is confusingly named and confusingly return typed, evidenced by how its callers inconsistently name the results (has_cudf, has_cuml, ..)

for it to be `lazy_import_has_dependecy_cuda`, i'd expect it to only check for something thin like `cupy`

i think you really mean `lazy_import_has_dependency_cu_cat()` and either it returns just `import cu_cat` or say `cudf, cuml, cu_cat`